apps/frontend/src/components/DifficultySelectPage.tsx

```diff
@@ -54,6 +62,25 @@ function DifficultySelectPage() {
       <Button variant="contained" onClick={matchUser}>
         Match
       </Button>
+
+      <Dialog open={isMatching} onClose={() => setIsMatching(false)}>
+        <DialogTitle>Matching User</DialogTitle>
+        <DialogContent>
+          <CountdownCircleTimer
+            isPlaying={isMatching}
+            duration={30}
+            colors={['#004777', '#F7B801', '#A30000', '#A30000']}
+            colorsTime={[25, 15, 5, 0]}
+          >
```

Comment:
add `onComplete` which calls `setIsMatching(false)` so that it closes after the 30 seconds

Reply:
Added